tests/Unit/NumericalAlgorithms/Integration/Test_GslQuad.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include <cmath>
+#include <cstddef>
+
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+/// [integrated_function]
+double gaussian(double x, double mean, double factor) {
+  return 2. * factor / sqrt(M_PI) * exp(-square(x - mean));
+}
+
+/// [integrated_function]
+double integrable_singularity(double x, double factor) {
+  return factor * cos(sqrt(abs(x))) / sqrt(abs(x));
+}
+
+SPECTRE_TEST_CASE("Unit.Numerical.Integration.GslQuadAdaptive",
+                  "[Unit][NumericalAlgorithms]") {
+
+{
+  INFO("StandardGaussKronrod");
+  // Construct the integration and give an example
+  /// [integration_example]
+  integrate::GslQuadAdaptive<integrate::IntegralType::StandardGaussKronrod,
+                             double, double> integration{20};
+  const double mean = 5.;
+  const double factor = 2.0;
+  const double lower_bound = -4.;
+  const double upper_bound = 10.;
+  integration.set_parameter<0>(mean);
+  integration.set_parameter<1>(factor);
+  integration.set_integrand(&gaussian);
+  auto result = integration(lower_bound, upper_bound, 1.e-10, 4);
+  /// [integration_example]
+  CHECK(result.value == approx(factor * erf(upper_bound - mean) -
+                                   factor * erf(lower_bound - mean)));
+}
+
+{
+  INFO("InfiniteInterval");
+  integrate::GslQuadAdaptive<integrate::IntegralType::InfiniteInterval, double,
+                             double> integration{20};
+  const double mean = 5.;
+  const double factor = 2.0;
+  integration.set_parameter<0>(mean);
+  integration.set_parameter<1>(factor);
+  integration.set_integrand(&gaussian);
+  auto result = integration(1.e-10);
+  CHECK(result.value == approx(2. * factor));
+}
+
+{
+  INFO("UpperBoundaryInfinite");
+  integrate::GslQuadAdaptive<integrate::IntegralType::UpperBoundaryInfinite,
+                             double, double> integration{20};
+  const double mean = 5.;
+  const double factor = 2.0;
+  const double lower_bound = -4.;
+  integration.set_parameter<0>(mean);
+  integration.set_parameter<1>(factor);
+  integration.set_integrand(&gaussian);
+  auto result = integration(lower_bound, 1.e-10);
+  CHECK(result.value == approx(factor * (1 - erf(lower_bound - mean))));
+}
+
+{
+  INFO("LowerBoundaryInfinite");
+  integrate::GslQuadAdaptive<integrate::IntegralType::LowerBoundaryInfinite,
+                             double, double> integration{20};
+  const double mean = 5.;
+  const double factor = 2.0;
+  const double upper_bound = 10.;
+  integration.set_parameter<0>(mean);
+  integration.set_parameter<1>(factor);
+  integration.set_integrand(&gaussian);
+  auto result = integration(upper_bound, 1.e-10);
+  CHECK(result.value == approx(factor * (1 + erf(upper_bound - mean))));
+}
+
+{
+  INFO("IntegrableSingularitiesPresent");
+  integrate::GslQuadAdaptive<
+      integrate::IntegralType::IntegrableSingularitiesPresent, double>
+      integration{20};
+  const double mean = 5.;
```

Comment:
This variable and a bunch of others are unused. You should get compiler warnings about them. Please remove.